scipy/stats/_hypotests.py

```diff
@@ -663,3 +665,241 @@ def somersd(x, y=None):
+    hypothesis :
+    - :math:`H_0 : p_2 \leq p_1`
+    - :math:`H_0 : p_2 \geq p_1`
+    - :math:`H_0 : p_2 = p_1`
+
+    Which can be computed using the ``alternative`` parameter with the
+    values : ``less``, ``greater`` or ``two-sided`` (default one)
+    respectively.
+
+    References
+    ----------
+    .. [1] `Mehta, Cyrus & Senchaudhuri, Pralay. (2003)`_. "Conditional versus Unconditional Exact Tests for Comparing Two Binomials". # noqa: E501
+
+    .. _`Mehta, Cyrus & Senchaudhuri, Pralay. (2003)`: https://www.statsols.com/hubfs/Resources_/Comparing-Two-Binomials.pdf # noqa: E501
+
+    Examples
+    --------
+    An example use of Barnard's test is presented in [1]_.
+
+        Consider the following example of a vaccine efficacy study
+        (Chan, 1998). In a randomized clinical trial of 30 subjects, 15 were
+        innoculated with a recombinant DNA influenza vaccine and the 15 were
+        innoculated with a placebo. Twelve of the 15 subjects in the placebo
+        group (80%) eventually became infected with influenza whereas for the
+        vaccine group, only 7 of the 15 subjects (47%) became infected. The
+        data are tabulated as a 2 x 2 table::
+
+                Vaccine  Placebo
+            Yes     7        12
+            No      8        3
+
+    When working with Statistical hypothesis testing, we usually use a
+    threshold probability or a significance level upon which we decide
+    to reject or not the null hypothesis :math:`H_0`. A commonly used
+    significance level is 5%.
+    To compute the p-value with Banard test, let's call `barnard_exact` as
+    follow :
+
+    >>> import scipy.stats as stats
+    >>> res = stats.barnard_exact([[7, 12], [8, 3]], alternative="less")
+    >>> res.statistic
+    1.894...
+    >>> res.pvalue
+    0.03407...
+
+    Thus, the probability of obtaining test results at least as extreme as the
+    data observed above is 3.4%. In this exemple, we are using the
+    ``alternative`` parameter with the "less" option, because the vaccine has
+    either no effect (:math:`H_0` hypothesis) or a positif effect
+    (:math:`H_1` hypothesis). Therefore, :math:`p_2` is either equal or lower
+    than `p_1` under the null hypothesis. Since our p-value is under our
+    significance level defined above, we reject :math:`H_0`. The vaccine has
+    a positive effect and we have :math:`p_1 \leq p_2` : The probability of
+    contracting the virus if we have been vaccinated is lower than if
+    we are not.
+
+    We can compare with Fisher's exact test which produces an exact
+    p-value of 6.4% :
+
+    >>> _,  pvalue= stats.fisher_exact([[7, 12], [8, 3]], alternative="less")
+    >>> pvalue
+    0.0640...
+
+    With a similar threshold probabilit of 5%, `fisher_exact` is accepting
+    :math:`H_0` while `barnard_exact` is rejecting it. As stated in [1]_,
+    Barnard test is more powerful than Fisher exact test.
```

Comment:
```suggestion
    Barnard's test is more powerful than Fisher's exact test.
```

Reply:
Update in d5633d95bed8764966e29add496e6c40c57a6f0c